layout-server/src/main/java/org/finos/vuu/layoutserver/controller/GlobalExceptionHandler.java

```diff
@@ -0,0 +1,44 @@
+package org.finos.vuu.layoutserver.controller;
+
+import java.util.List;
+import java.util.NoSuchElementException;
+import java.util.stream.Collectors;
+import org.springframework.http.ResponseEntity;
+import org.springframework.http.converter.HttpMessageNotReadableException;
+import org.springframework.web.bind.MethodArgumentNotValidException;
+import org.springframework.web.bind.annotation.ControllerAdvice;
+import org.springframework.web.bind.annotation.ExceptionHandler;
+import org.springframework.web.method.annotation.MethodArgumentTypeMismatchException;
+
+@ControllerAdvice
+public class GlobalExceptionHandler {
+
+    @ExceptionHandler(NoSuchElementException.class)
+    public ResponseEntity<String> handleNotFound(NoSuchElementException ex) {
+        return new ResponseEntity<>(ex.getMessage(), org.springframework.http.HttpStatus.NOT_FOUND);
+    }
+
+    @ExceptionHandler(MethodArgumentNotValidException.class)
+    public ResponseEntity<String> handleMethodArgumentNotValid(MethodArgumentNotValidException ex) {
+        List<String> errors = ex.getFieldErrors().stream()
+            .map(fieldError -> fieldError.getField() + ": " + fieldError.getDefaultMessage())
+            .collect(Collectors.toList());
+
+        return new ResponseEntity<>(errors.toString(),
+            org.springframework.http.HttpStatus.BAD_REQUEST);
+    }
+
+    @ExceptionHandler(MethodArgumentTypeMismatchException.class)
+    public ResponseEntity<String> handleMethodArgumentTypeMismatchException(
+        MethodArgumentTypeMismatchException ex) {
+        return new ResponseEntity<>(ex.getMessage(),
+            org.springframework.http.HttpStatus.BAD_REQUEST);
+    }
+
+    @ExceptionHandler(HttpMessageNotReadableException.class)
+    public ResponseEntity<String> handleHttpMessageNotReadableException(
+        HttpMessageNotReadableException ex) {
+        return new ResponseEntity<>(ex.getMessage(),
+            org.springframework.http.HttpStatus.BAD_REQUEST);
+    }
```

Comment:
I originally had that, and then separated it out (in [this commit](https://github.com/ScottLogic/finos-vuu/pull/68/commits/7dd8233feeb0442ac1db22ded9df90f86f54b6e0)) in an effort to improve the error messages, where `MethodArgumentNotValidException` needed to handle the error message slightly differently.

It could look confusing if we have a generic `handleBadRequest` exception handler when it isn't the only exception handler returning a bad request response. But, I do have a generic `handleNotFound`, so arguably that's hypocritical.

I think if we have the 'generic' exception handlers at the top, and any individual exception handlers below, I can get on board with this suggestion.